### PR TITLE
12 school

### PR DESCRIPTION
**Pull request by nisarg-serpentcs**
improvement for middle name modified removed and student roll no. duplication issue solved, and also Student name gets tree view and form view time full name displayed.

### CODE REVIEW

File: school/models/student.py [hunk 1]
```diff
@@ -24,6 +24,22 @@ class StudentStudent(models.Model):
     _name = 'student.student'
     _table = "student_student"
     _description = 'Student Information'
+    _rec_name = 'full_name'
+
+    @api.depends('stu_name', 'middle', 'last')
+    def _compute_full_name(self):
+        for rec in self:
+            if rec.middle:
```
Comment (AnuKPatel) [inline school/models/student.py]: This can be improved.

**Comment by nisarg-serpentcs** [comment]
changed method for _compute_full_name and remove if condition.

File: school/models/student.py [hunk 2]
```diff
@@ -268,6 +261,8 @@ def check_current_year(self):
                                      )
     active = fields.Boolean(default=True)
 
+    _sql_constraints = [('roll_no_key', 'UNIQUE (roll_no,standard_id,school_id)',  'Student Roll Numer. must be unique.!')]
+
```
Comment (AnuKPatel) [inline school/models/student.py]: Spelling mistake

File: school/models/student.py [hunk 3]
```diff
@@ -29,17 +29,11 @@ class StudentStudent(models.Model):
...
-                partner_name += " " + rec.last if rec.last else ""
-                rec.full_name = partner_name
-            else:
-                partner_name = ""
-                partner_name += rec.stu_name if rec.stu_name else ""
-                partner_name += " " + rec.last if rec.last else ""
-                rec.full_name = partner_name
+            partner_name = ""
+            partner_name += rec.stu_name if rec.stu_name else ""
+            partner_name += " " + rec.middle if rec.middle else ""
```
Comment (AnuKPatel) [inline school/models/student.py]: " " will add in partner_name no matter middle name is there or not.

**Comment by nisarg-serpentcs** [comment]
Improve for _compute_full_name method and removed condition inline.

**Comment by nisarg-serpentcs** [comment]
spelling mistake changed

File: school_evaluation/models/school_evaluation.py [hunk 4]
```diff
@@ -152,6 +152,7 @@ class SchoolEvaluationTemplate(models.Model):
                             'User Type', required=True, default='faculty',
                             help="Type of Evaluation")
     rating_line = fields.One2many('rating.rating', 'rating_id', 'Rating')
```
Comment (AnuKPatel) [inline school_evaluation/models/school_evaluation.py]: this field should be removed now

**Pull request closed without merging**